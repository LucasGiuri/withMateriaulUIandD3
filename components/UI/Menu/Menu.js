import PropTypes from 'prop-types';
import NavLink from '../NavLink/NavLink';
import { Ul, Li, Nav } from './Menu.styles';

const Menu = ({urls, currentPath}) => (
  <Nav>
    <Ul>
      {urls.map(link => {
        const { url, href, text } = link;
        return (
          <Li key={url} isActive={url === currentPath}>
            <NavLink href={href} as={url}>
              {text}
            </NavLink>
          </Li>
        )
      })}
    </Ul>
  </Nav>
);

Menu.propTypes = {
  urls: PropTypes.arrayOf(PropTypes.object),
  currentPath: PropTypes.string.isRequired,
};

Menu.defaultProps = {
  urls: [{}],
};

export default Menu;