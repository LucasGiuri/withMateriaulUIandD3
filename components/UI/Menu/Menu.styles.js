import styled from 'styled-components';
import { PRIMARY, SECONDARY, WHITE } from '../../../public/styles/colors';

export const Nav = styled.div`
  width: 100%;
  background: black;
`;
  
export const Ul = styled.div`
  display: flex;
  height: 40px;
  justify-content: space-around;
`;
  
export const Li = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  border-bottom: 2px solid ${props => props.isActive ? PRIMARY: SECONDARY};
  color: ${WHITE};
`;