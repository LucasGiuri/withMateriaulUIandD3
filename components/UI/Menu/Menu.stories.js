import React from 'react';
import { storiesOf } from '@storybook/react';
import Menu from './Menu';

const urls = [{
  url: "/",
  href: "/",
  text: "People"
},
{
  url: "/films",
  href: "/films",
  text: "Films"
},
{
  url: "/vehicles",
  href: "/vehicles",
  text: 'Vehicles'
},
{
  url: '/planets',
  href: '/planets',
  text: 'Planets'
}];

const currentPath = "/";

storiesOf('Menu')
  .add('Menu default', () => (
    <Menu urls={urls} currentPath={currentPath} />
  ))