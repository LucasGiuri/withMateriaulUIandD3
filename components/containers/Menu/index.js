import { withRouter } from 'next/router';
import Menu from '../../UI/Menu/Menu';

const MenuContainer = ({router}) => {
  const { pathname } = router;

  const urls = [{
    url: "/",
    href: "/",
    text: "People"
  },
  {
    url: "/films",
    href: "/films",
    text: "Films"
  },
  {
    url: "/vehicles",
    href: "/vehicles",
    text: 'Vehicles'
  },
  {
    url: '/planets',
    href: '/planets',
    text: 'Planets'
  }];

  return <Menu urls={urls} currentPath={pathname} />
};

export default withRouter(MenuContainer);