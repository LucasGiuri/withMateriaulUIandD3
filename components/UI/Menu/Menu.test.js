import Menu from './Menu';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const urls = [{
  url: "/",
  href: "/",
  text: "People"
},
{
  url: "/films",
  href: "/films",
  text: "Films"
},
{
  url: "/vehicles",
  href: "/vehicles",
  text: 'Vehicles'
},
{
  url: '/planets',
  href: '/planets',
  text: 'Planets'
}];

const currentPath = "/";

describe('Menu', () => {
	it('Menu basic render snapshoot with empty list', () => {
		const component = renderer.create(
      <Menu urls={urls} currentPath={currentPath} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})