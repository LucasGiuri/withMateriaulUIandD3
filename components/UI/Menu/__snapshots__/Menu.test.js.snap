// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu basic render snapshoot with empty list 1`] = `
.c3 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Roboto',sans-serif;
}

.c0 {
  width: 100%;
  background: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 2px solid #DC3545;
  color: #FFFFFF;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 2px solid rgb(1,108,249);
  color: #FFFFFF;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        className="c3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        People
      </a>
    </div>
    <div
      className="c4"
    >
      <a
        className="c3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Films
      </a>
    </div>
    <div
      className="c4"
    >
      <a
        className="c3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Vehicles
      </a>
    </div>
    <div
      className="c4"
    >
      <a
        className="c3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Planets
      </a>
    </div>
  </div>
</div>
`;
